### ISSUE TITLE
$.ajax({ dataType: 'script' }) pending indefinitely in case of error on jQuery 1.x

### ISSUE THREAD

**Issue by Krinkle** [jquery/jquery#2413]
With jQuery 1.x, when a script request has some kind of error (User not online, DNS can't be resolved, server responds with HTTP 503, etc.) the Promise is never resolved or rejected, it stays pending indefinitely.

In jQuery 2.x it gets instantly rejected.

This can be worked around by attaching a timeout, but users tend not to do that. This causes lots of components to stall indefinitely when certain problems occur with the server, or the user's connection.

Pending indefinitely when using jQuery 1.x on latest Chrome. Be sure to test on an HTTP origin as otherwise Chrome might prevent the request for security reasons (though those errors don't propagate either).
```js
 $.ajax({ url: 'http://unknown.jquery.com/foo', dataType: 'script', cache: true })
  .then(function () { console.log( 'done' ); }, function () { console.log( 'fail' ); });

 $.ajax({ url: 'http://example.org/404', dataType: 'script', cache: true })
  .then(function () { console.log( 'done' ); }, function () { console.log( 'fail' ); });
```

**Comment by mzgol** [comment]
Confirmed on `compat`: http://jsfiddle.net/tvqs3t6o/2/ on Firefox 38.0.5 & Chrome 43. Works fine on `master`.

**Comment by markelog** [comment]
`compat-only` sounds analogues to the `blocker` label. 
@timmywil am i right?

**Comment by mzgol** [comment]
What do you mean? I don't see an analogy.

`compat-only` means the issue affect the `compat` branch only. There's also
a similar `master-only` label.

**Comment by markelog** [comment]
Can we release if we have behaviour discrepancy between two versions?

**Comment by mzgol** [comment]
Ah, this. Some of these are just edge cases, some are internal modifications that don't affect public API (see e.g. #2133 or #1710), some affect IE8 only so are `compat-only` by definition (e.g. #1745). So I don't think `compat-only` implies `blocker`, far from it.

If an issue indicates serious API incompatibility between versions then that may implictate a blocker but it's IMO not automatic.

**Comment by markelog** [comment]
Okay, reasonable, but this is a `blocker`?

**Comment by timmywil** [comment]
I wouldn't call this a blocker for alpha, but it should be fixed before the final release.

**Comment by mzgol** [comment]
@timmywil Since this is not fixed on `compat` and we don't plan 3.0.0 Compat anymore, I'm closing it. Re-open if you disagree.
